src/qt/rpcconsole.cpp

```diff
@@ -464,6 +446,101 @@ void RPCExecutor::request(const QString &command, const WalletModel* wallet_mode
     }
 }
 
+/**
+ * Small and fast parser supporting console command syntax, with limited functionality.
+ * Splits a command line string into a vector with command at position 0, and parameters after
+ *
+ * @param[in]    strCommand  Command line to parse
+ *
+ * @return a vector of strings with command and parameters
+ */
+
+std::vector<std::string> RPCExecutor::parseHelper(const std::string& strCommand)
+{
+    std::vector<std::string> vec = SplitString(strCommand, " (),\n");
+    auto should_remove = [](const std::string& str) { return str.empty(); };
+    vec.erase(std::remove_if(vec.begin(), vec.end(), should_remove), vec.end());
+    return vec;
+}
+
+/**
+ * Catches console-only command before a RPC call is executed
+ *
+ * @param[in]    command  Command line to execute
+ * @param[in]    wallet_model  Wallet model to use
+ * @return  true if command was handled by this method (even on errors), false otherwise
+ *
+ */
+
+bool RPCExecutor::executeConsoleOnlyCommand(const std::string& command, const WalletModel* wallet_model)
+{
+    std::string result;
+    std::string address;
+    std::string executableCommand;
+    std::vector<std::string> parsedCommand{parseHelper(command)};
+
+    // Catch the console-only generate command with 2 or less parameters before RPC call is executed .
+    if (!parsedCommand.empty() && parsedCommand[0] == "generate" && parsedCommand.size() <= 3) {
+        // Default number of blocks is 1 if missing
+        const std::string nblocks{parsedCommand.size() > 1 ? parsedCommand[1] : "1"};
+        // An empty maxtries parameters will be modified by generatetoaddress' default value
+        const std::string maxtries{parsedCommand.size() > 2 ? parsedCommand[2] : ""};
+
+        // Generate address
+        if (!RPCConsole::RPCExecuteCommandLine(m_node, address, "getnewaddress\n", nullptr, wallet_model)) {
+            Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Error: could not generate new address"));
+        } else {
+            executableCommand = "generatetoaddress " + nblocks + " \"" + address + "\" " + maxtries;
+            if (!RPCConsole::RPCExecuteCommandLine(m_node, result, "generatetoaddress " + nblocks + " " + address + " " + maxtries + "\n", nullptr, wallet_model)) {
```

Comment:
```suggestion
            auto executableCommand = "generatetoaddress " + nblocks + " " + address + " " + maxtries;
            if (!RPCConsole::RPCExecuteCommandLine(m_node, result, executableCommand + "\n", nullptr, wallet_model)) {
```
Also remove the declaration of `executableCommand` near the top of the function; better to declare variables only where and when needed. You could do similarly with `result` and `address`.

Reply:
I ' ll actually remove the variable `executableCommand` since it's only used once, I prefer avoiding memory overhead and better speed execution (althoug minimal). Okey with the other 2 variables